### PR TITLE
refactoring UnitElimination

### PR DESCRIPTION
**Pull request by madvorak**

### CODE REVIEW

File: PumpingCfg/UnitElimination.lean [hunk 1]
```diff
@@ -115,155 +110,153 @@ lemma generators_prod_diag_subset : g.generators_prod_diag ⊆ g.generators ×ˢ
...
-lemma collect_unitPairs_unitPair {r : ContextFreeRule T g.NT} (pairs : List (g.NT × g.NT))
-    (hmem : r ∈ g.rules) (hp : ∀ p ∈ pairs, UnitPair p.1 p.2) :
-    ∀ p ∈ collect_unitPairs r pairs, UnitPair p.1 p.2 := by
-  intro p h'
-  unfold collect_unitPairs at h'
-  match heq : r.output with
+        rw [hpv]
+        exact ⟨Or.inr hvl, rfl⟩
+
+lemma collect_unitPairs_unitPair {r : ContextFreeRule T g.NT} (S : List (g.NT × g.NT))
```
Comment (AlexLoitzl) [inline PumpingCfg/UnitElimination.lean:204]: Why do you call Lists and Finsets L/S

Comment (AlexLoitzl) [reply PumpingCfg/UnitElimination.lean:204]: And should I also do this at EpsilonElimination

Comment (madvorak) [reply PumpingCfg/UnitElimination.lean:204]: Well, I'm kinda wannabe mathematicians, so I guess I just like naming data variables by single letters.

Comment (madvorak) [reply PumpingCfg/UnitElimination.lean:204]: Lowercase is probably better tho.

**Pull request merged**